### PR TITLE
Development

### PR DESCRIPTION
**Pull request by NeerNim**

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+describe "Board Class" do
+  let(:board) {Board.new}
+
+  it "should have board class" do  
+    expect(board.class).to eql(Board)
+  end
+  
+  describe "#update_board" do   
+    it "should change the board cell number to character" do
+      expect(board.update_board(1, "X")).to eql(board.board[0])
```
Comment (mcrd25) [inline spec/board_spec.rb:12]: You should do a test for incorrect number and number in space already taken.

That should give you 2 more it statements

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  attr_accessor :continue
+
+  def initialize(board, player1, player2)
+    @new_board = board
+    @player1 = player1
+    @player2 = player2
+  end
+
+  def play
```
Comment (mcrd25) [inline lib/game.rb:17]: This method was not tested in your game_spec.rb

I am sure you can think of some ways of testing this method.

To consider project complete I must see that you have tested all public  methods of classes you have implemented in this project.


I'm sure you can think of some 😄! If not, I am here to help, simply contact me!

File: lib/player.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+# class Player
+class Player
+  include UI
+
+  attr_reader :sign
+  def initialize(sign)
+    @sign = sign
+  end
+
+  def make_move(board)
```
Comment (mcrd25) [inline lib/player.rb:14]: As mentioned before, all public methods should be tested. 
Create player_spec.rb file to test this method.

Thanks 😄

File: lib/ui.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  DRAWMSG = "It's a draw\nWould you like to play again? Type Y/N"
+  def instructions
+    puts 'Tictactoe game is 3X3 board game. You need to choose either X or O'
+    puts '******************************************************************'
+    puts '---Every turn the players puts the sign X or O on their turn if---'
+    puts '******************************************************************'
+    puts '----Any player gets 3 consequetive X or O then he or she wins-----'
+  end
+
+  def show_board(board)
```
Comment (mcrd25) [inline lib/ui.rb:16]: I'm sure you guys can find a way to test this method.

Hint: Try testing stdout expected

File: lib/ui.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    puts UI::DRAWMSG
+    play_again?
+  end
+
+  def player2_msg
+    puts UI::PLAYER2MSG
+    play_again?
+  end
+
+  def display_message
```
Comment (mcrd25) [inline lib/ui.rb:81]: test this method please :)

File: spec/ui_spec.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+  describe "#choose_num" do
+    it "returns the chosen cell if only its available" do
+      board.update_board(3, "X")
+      stub(:gets) {"2"}
+      expect(choose_num(board)).to eql("2")
+    end
+  end
+
+  describe "#choose_sign" do
```
Comment (mcrd25) [inline spec/ui_spec.rb:44]: Maybe you can try testing for when Incorrect Sign value is given.

Contact me and maybe we can brainstorm together =).

For now though, this is not required

File: spec/ui_spec.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+
+  describe "#play_again" do
+    it "takes an answer as Y or N and returns it" do
+      stub(:gets) {"N"}
+      expect(play_again?).to eql('N')
+    end
+  end
+
+  describe "#choose_num" do
```
Comment (mcrd25) [inline spec/ui_spec.rb:24]: Should have at least one test for when an out of bounds value is given. I strongly recommend that you do, as this is easy to test.

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# WOW! I'm really impressed! You guys have done a great job. You are ALMOST ready for taking off 🚀 

- **Unfortunately, I cannot give you the 👍 just yet as you have not tested all public methods as yet. This is one of the requirements.
Please also take a look at review comments on certain existing tests you can improve!**

- **It would also be cool if you can add a live version of the original Tic-Tac-Toe project and the repo. Your README.md is excellent though!**

***If you have any questions, let me know here or on slack and I will be glad to clarify with you.***

**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/)

File: spec/board_spec.rb (hunk 1 above)
Comment (YemreAybey) [reply spec/board_spec.rb:12]: Hey Maya, our update board method doesn't check if the given number is correct or if the cell is already taken, there are other methods which do that and we are testing them.
`def check_sign(num) `
     `@board[num.to_i - 1] == num.to_i`
 `end`
this method, for instance, is checking if the cell is already taken.
so we are confused about why we should test update_board method with incorrect number and number in space already taken.

File: lib/game.rb (hunk 2 above)
Comment (NeerNim) [reply lib/game.rb:17]: The change for this method has been done

File: lib/player.rb (hunk 3 above)
Comment (NeerNim) [reply lib/player.rb:14]: The change for this method has been done

File: lib/ui.rb (hunk 4 above)
Comment (NeerNim) [reply lib/ui.rb:16]: The change for this method has been done

File: lib/ui.rb (hunk 5 above)
Comment (NeerNim) [reply lib/ui.rb:81]: The change for this method has been done

File: spec/ui_spec.rb (hunk 6 above)
Comment (NeerNim) [reply spec/ui_spec.rb:44]: The change for this method has been done

File: spec/ui_spec.rb (hunk 7 above)
Comment (NeerNim) [reply spec/ui_spec.rb:24]: The change for this method has been done

File: spec/board_spec.rb (hunk 1 above)
Comment (mcrd25) [reply spec/board_spec.rb:12]: That's fine.

Misunderstanding on my part.

**Review by mcrd25 — APPROVED** [review]
# Great Job doing the tests! Your Version of Tic Tac Toe was also different from what I've seen so far! good job! 💯 🥇 👍 

*YOU MAY NOW MERGE TO MASTER!* 🚀 

**YOU GUYS DID A GREAT JOB WITH THE TESTS! I'M IMPRESSED!**

***If you have any questions, now or in the future, do not hesitate to contact me on social media or on slack and I will be glad to chat with you.***
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/)

**Pull request merged**